delivery/services/delivery_service.py

```diff
@@ -79,40 +89,54 @@ def deliver_by_staging_id(self, staging_id, delivery_project, md5sum_file):
 
         args_for_run_mover = {'delivery_order_id': delivery_order.id,
                               'delivery_order_repo': self.delivery_repo,
-                              'external_program_service': self.external_program_service,
+                              'external_program_service': self.mover_external_program_service,
                               'session_factory': self.session_factory}
 
-        self.io_loop_factory().spawn_callback(MoverDeliveryService._run_mover,
-                                              **args_for_run_mover)
+        if skip_mover:
+            session = self.session_factory()
+            delivery_order.delivery_status = DeliveryStatus.delivery_skipped
+            session.commit()
+        else:
+            self.io_loop_factory().spawn_callback(MoverDeliveryService._run_mover,
+                                                  **args_for_run_mover)
 
         return delivery_order.id
 
+    @staticmethod
+    def _parse_status_from_mover_info_result(mover_info_result):
+        #Parse status from this type of example string:
+        # Delivered: Jan 19 00:23:31 [1484781811UTC]
+        pattern = re.compile('^(\w+):\s')
+        hits = pattern.match(mover_info_result)
+        if hits:
+            return hits.group(1)
+        else:
+            raise CannotParseMoverOutputException("Could not parse mover id from: {}".format(mover_info_result))
```

Comment:
Error message seems slightly off - should it be something like `"Could not parse mover status for delivery id {} from: {}"`